File: src/grap3fruit/programmers/42578.py
import itertools

clothes = [["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"], ["bluen_turban", "headgear"], ["blue_sung1", "eyewear"], ["blue1g1", "shoes"], ["blue_1", "shoes"]]

answer = 0
bucket = {}

for cloth in clothes:
	cloth_type = cloth[1]
	if not cloth_type in bucket :
		bucket[cloth_type] = 1
	
	elif cloth_type in bucket :
		bucket[cloth_type] += 1

arr = list(bucket.values())

result = 0
for i in range(1, len(arr)+1):
	nCr = itertools.combinations(arr, i)
	calc_arr = list(nCr)
	print(calc_arr)
	for j in range(0, len(calc_arr)):
		multi = 1
		for el in calc_arr[j] :
			multi *= el
		
		result += multi

print(result)

File: src/grap3fruit/js/goorm/42578.js
function solution(clothes) {
  let answer = 1;
  const data = {};

  clothes.forEach((cloth) => {
    if (!data[cloth[1]]) {
      return (data[cloth[1]] = [cloth[0]]);
    }
    data[cloth[1]].push(cloth[0]);
  });

  for (const el in data) {
    answer *= data[el].length + 1;
  }
  answer -= 1;

  return answer;
}

const clothes = [
  ['yellow_hat', 'headgear'],
  ['blue_sunglasses', 'eyewear'],
  ['green_turban', 'headgear'],
];

console.log(solution(clothes));
